deltacat/compute/compactor/compaction_session.py

```diff
@@ -90,6 +90,7 @@ def compact_partition(
     ] = None,  # TODO (ricmiyam): Remove this and retrieve schema from storage API
```

Comment:
we would also want to track head node memory usage (which is more important) as otherwise we'd have to vertically scale.